lib/SIL/SILPrinter.cpp

```diff
@@ -3097,26 +3142,32 @@ void SILDifferentiabilityWitness::print(
     if (!requirements.empty()) {
       OS << " [where ";
       auto subPrinter = PrintOptions::printSIL();
+      subPrinter.GenericEnv = origGenEnv;
       interleave(requirements,
                  [&](Requirement req) {
                    req.print(OS, subPrinter);
                  },
                  [&] { OS << ", "; });
-      OS << ']';
+      OS << "] ";
     }
   }
   // @original-function-name : $original-sil-type
-  OS << " @" << originalFunction->getName() << " : "
-     << originalFunction->getLoweredType();
+  printSILFunctionNameAndType(OS, originalFunction);
   // {
   //   jvp: @jvp-function-name : $jvp-sil-type
   //   vjp: @vjp-function-name : $vjp-sil-type
   // }
   OS << " {\n";
-  if (jvp)
-    OS << "  jvp: @" << jvp->getName() << " : " << jvp->getLoweredType() << '\n';
-  if (vjp)
-    OS << "  vjp: @" << vjp->getName() << " : " << vjp->getLoweredType() << '\n';
+  if (jvp) {
+    OS << "  jvp: ";
+    printSILFunctionNameAndType(OS, jvp);
+    OS << "\n";
+  }
+  if (vjp) {
+    OS << "  vjp: ";
+    printSILFunctionNameAndType(OS, vjp);
+    OS << "\n";
```

Comment:
Use character literal.

Reply:
Done in https://github.com/apple/swift/pull/27652/commits/fa71b2609d18a32ac158ae7b9e7e7f0dc013df90.

Reply:
Same here.